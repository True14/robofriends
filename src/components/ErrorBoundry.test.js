import React from "react";
import { shallow } from "enzyme";
import ErrorBoundry from "./ErrorBoundry";

describe("when no JS errors are caught in a child component", () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallow(
      <ErrorBoundry>
        <h1>hello</h1>
      </ErrorBoundry>
    );
  });

  it("should render", () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe("when JS errors are caught in a child component", () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallow(
      <ErrorBoundry>
        <h1>hello</h1>
      </ErrorBoundry>
    );
    wrapper.instance().componentDidCatch("oh no an error!");
    wrapper.update();
  });

  it("should update the state to indicate an error", () => {
    expect(wrapper.instance().state.hasError).toEqual(true);
  });
});
