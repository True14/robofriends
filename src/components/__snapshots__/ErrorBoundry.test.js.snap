// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when no JS errors are caught in a child component should render 1`] = `
<h1>
  hello
</h1>
`;
